File: Web-Frontend/src/api/messageApi.js
import API from "./axios";



export const getChatUsers = () => API.get("/messages/users");

export const getConversations = () => API.get("/messages/conversations");

export const getOrCreateDirectConversation = (userId) =>
  API.post("/messages/conversations/direct", { userId });

export const createGroupConversation = (name, participants) =>
  API.post("/messages/conversations/group", { name, participants });

 
// update group name and/or add members
export const updateGroupConversation = (conversationId, data) =>
  API.patch(`/messages/conversations/${conversationId}`, data);

export const getMessages = (conversationId) => API.get(`/messages/${conversationId}`);

// fetch shared files/media/audio for a conversation
export const getSharedMedia = (conversationId) =>
  API.get(`/messages/${conversationId}/shared-media`);

export const sendMessage = (conversationId, formData) =>
  API.post(`/messages/${conversationId}`, formData, {
    headers: { "Content-Type": "multipart/form-data" },
  });

export const deleteMessage = (messageId) =>
  API.delete(`/messages/message/${messageId}`);

File: Mobile-App/mint_mobile/lib/api/message_api.dart
import 'package:dio/dio.dart';
import 'dio_client.dart';

class MessageApi {
  static Future<Response> getChatUsers() =>
      DioClient.instance.get('/messages/users');

  static Future<Response> getConversations() =>
      DioClient.instance.get('/messages/conversations');

  static Future<Response> getOrCreateDirect(String userId) =>
      DioClient.instance.post(
        '/messages/conversations/direct',
        data: {'userId': userId},
      );

  static Future<Response> createGroup(
          String name, List<String> participants) =>
      DioClient.instance.post(
        '/messages/conversations/group',
        data: {'name': name, 'participants': participants},
      );

  static Future<Response> getMessages(String conversationId) =>
      DioClient.instance.get('/messages/$conversationId');

  /// Sends a message (text or file) to a conversation.
  ///
  /// [formData] must already contain either a 'text' field or a 'file'
  /// MultipartFile (or both for a caption).
  ///
  /// [onSendProgress] is optional — pass it from ChatProvider.sendFile
  /// to track upload progress (bytes sent / total bytes).
  static Future<Response> sendMessage(
    String conversationId,
    FormData formData, {
    void Function(int sent, int total)? onSendProgress,
  }) =>
      DioClient.instance.post(
        '/messages/$conversationId',
        data: formData,
        onSendProgress: onSendProgress,
        options: Options(
          contentType: 'multipart/form-data',
          // Give large file uploads up to 5 minutes before timing out.
          // Adjust to match your server's own upload limit.
          sendTimeout: const Duration(minutes: 5),
          receiveTimeout: const Duration(minutes: 2),
        ),
      );

  static Future<Response> deleteMessage(String messageId) =>
      DioClient.instance.delete('/messages/message/$messageId');

  /// Edit a sent message.
  /// Uses PUT — change to .patch() if your backend uses PATCH instead.
  static Future<Response> editMessage(
          String messageId, String newText) =>
      DioClient.instance.put(
        '/messages/message/$messageId',
        data: {'text': newText},
      );
}